src/classes/RD2_OpportunityEvaluationService_TEST.cls

```diff
@@ -1891,6 +1891,154 @@ private class RD2_OpportunityEvaluationService_TEST {
             'No Opportunity should be created when the Disable First Installment is true');
     }
 
+    /**
+    * @description Verifies that when RD's next installment is skipped(paused),
+    * the RD status will be updated to Pause when the nightly batch job runs
+    */
+    @isTest
+    private static void shouldUpdateRDToPauseStatusWhenNextInstallmentIsPaused() {
+        Date today = Date.newInstance(2020, 1, 1);
+        Date pauseDate = Date.newInstance(2020, 2, 1);
+        RD2_ScheduleService.currentDate = today;
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationBuilder(getContact().Id)
+            .withDayOfMonth('1')
+            .build();
+
+        insert rd;
+
+        RD2_ScheduleService_TEST.createPauseSchedule(rd.Id, pauseDate, pauseDate);
+
+        Test.startTest();
+        new RD2_OpportunityEvaluationService()
+            .withRds(new Set<Id>{ rd.Id })
+            .withCurrentDate(today.addDays(10))
+            .evaluateOpportunities();
+        Test.stopTest();
+
+        rd = rdGateway.getRecord(rd.Id);
+
+        System.assertEquals(RD2_Constants.STATUS_PAUSED, rd.Status__c,
+            'Rd status should be paused when the next installment is paused');
+        System.assertEquals(RD2_ScheduleService_TEST.PAUSE_REASON, rd.ClosedReason__c,
+            'The pause reason shuold be updated according to the user selection');
+    }
+
+    /**
+    * @description Verifies that when the RD is in the middle of pause period
+    * the RD status should stay as Pause when the nightly batch job runs
```

Comment:
since we're switching to "Paused", comments should probably reflect that as well.